src/main/java/org/mule/extension/ftp/internal/FtpUtils.java

```diff
@@ -35,6 +35,11 @@ public static String normalizePath(String path) {
     return FilenameUtils.normalize(path, true);
   }
 
+  public static String getReplyErrorMessage(Integer replyCode, String replyString) {
+    return FTPReply.isPositiveCompletion(replyCode) ? ""
+        : getReplyCodeErrorMessage(replyCode) + format(". FTP reply string is: %s", replyString);
```

Comment:
Method `getReplyCodeErrorMessage(...)` checks the same condition again. I prefer repeat message formatting instead of checking that condition twice.
```
return FTPReply.isPositiveCompletion(replyCode) ? ""
        : format("FTP reply code is: %d. FTP reply string is: %s", replyCode, replyString);
```

Reply:
fixed